static/js/popover_menus.js

```diff
@@ -51,25 +51,31 @@ function on_show_prep(instance) {
 }
 
 export function initialize() {
-    delegate("body", {
-        ...default_popover_props,
-        target: "#streams_inline_cog",
-        onShow(instance) {
-            on_show_prep(instance);
-            instance.setContent(
-                render_left_sidebar_stream_setting_popover({
-                    can_create_streams:
-                        settings_data.user_can_create_private_streams() ||
-                        settings_data.user_can_create_public_streams() ||
-                        settings_data.user_can_create_web_public_streams(),
-                }),
-            );
-            left_sidebar_stream_setting_popover_displayed = true;
-        },
-        onHidden() {
-            left_sidebar_stream_setting_popover_displayed = false;
-        },
-    });
+    if (
+        settings_data.user_can_create_private_streams() ||
+        settings_data.user_can_create_public_streams() ||
+        settings_data.user_can_create_web_public_streams()
+    ) {
```

Comment:
Would it be possible to skip this conditional and just have the logic in the other function do `e.stopPropagation` to prevent the popover from being opened?  That would feel cleaner in avoiding having two copies of this somewhat involved conditional that need to be synchronized for correct behavior, and would also prevent a bug wherein this UI widget won't work until a browser reload if a user is granted stream creation permission.

(Because this initialization code wouldn't be called again).

Reply:
@timabbott, Thank you for the response.
Altough I was assuming `e.stopPropagation` to not work as both, the `onClick` event and tippy `delegate` were attached to the same heirarchy level element(**#streams_inline_cog**) but it seems to be working anyway. Will update my PR according to the requirements.

Reply:
Also, I don't think it would be possible to execute this functioning without the use of that conditional. It would have to be used once.